File: AHUR2/A.py
import math

MOD = 998244353

T = int(input())


def qpow(base: int, q: int):
    ans = 1
    while q:
        if q & 1:
            ans *= base
            ans %= MOD
        q >>= 1
        base *= base
        base %= MOD
    return ans


for _ in range(T):
    n, k = list(map(int, input().split()))
    arr = [0] * int(1e3 + 1)
    for i in range(1001):
        arr[i] = qpow(i, k)
    dp = [0] * int(1e6 + 1)
    for i in range(int(1e6), int(1e3), -1):
        dp[i] += arr[int(1e6) // i]  # = 1e6 / i ** k
        # for j in range(1, int(math.sqrt(i)) + 1, 1):
        #     if i % j == 0:
        #         if j != 1:
        #             dp[i // j] -= dp[i]
        #         if j * j != i:
        #             dp[i // (i // j)] -= dp[i]
        for j in range(1, int(1e6 + 1)):
            if j * i > int(1e6):
                break
            dp[i] -= dp[i * j]

    for i in range(int(1e3), 0, -1):
        dp[i] += qpow(int(1e6) // i, k)
        # for j in range(1, int(math.sqrt(i)) + 1, 1):
        #     if i % j == 0:
        #         if j != 1:
        #             dp[i // j] -= dp[i]
        #         if j * j != i:
        #             dp[i // (i // j)] -= dp[i]
        for j in range(1, int(1e6 + 1)):
            if j * i > int(1e6):
                break
            dp[i] -= dp[i * j]

    ans = 1
    for i in range(int(n + 1)):
        ans *= qpow(i, dp[i])
        ans %= MOD
    print(ans)

    # 对于i > 1e3，可以预处理
    # i <= 1e3，则只计算

File: AHUR2/A.cpp
/**
 *@author ChatGPT（基于 GPT-4o-mini）
 */
// This code was translated and adapted by ChatGPT, an AI language model based on GPT-4o-mini.
// The assistant's role was to convert Python logic into equivalent C++ code,
// focusing on preserving functionality while maintaining clarity and performance.
// Python’s ease is great for prototyping, but damn, it can seriously drag you down when it counts.

#include <bits/stdc++.h>
using namespace std;
using i64 = int64_t;

const i64 MOD  = 998244353;
const i64 MOD2 = 998244352;

struct Prime {
    vector<int> minp, primes;

    Prime(int n)
    {
        minp.assign(n + 1, 0);
        for (int i = 2; i <= n; i++) {
            if (minp[i] == 0) {
                minp[i] = i;
                primes.push_back(i);
            }
            for (int p : primes) {
                if ((i * 1LL * p) > n) break;
                minp[i * p] = p;
                if (p == minp[i]) break;
            }
        }
    }
};

i64 qpow(i64 base, i64 exp, i64 mod)
{
    i64 res = 1;
    base %= mod;
    while (exp > 0) {
        if (exp & 1) res = (res * base) % mod;
        base = (base * base) % mod;
        exp >>= 1;
    }
    return res;
}

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    Prime prime(int(1e6));

    int T;
    cin >> T;
    while (T--) {
        string K;
        i64    n, k;
        cin >> n >> K;
        k       = 0;
        int len = K.length();
        for (int i = 0; i < len; i++)
            k = ((10ll * k + K[i] - '0') + 402653184) % 402653184;

        i64         ans = 1;
        vector<int> buffer;
        int         sqrtN = (int)sqrtl(n);

        for (auto p : prime.primes) {
            if (p <= sqrtN) {
                i64 q = 0;
                i64 a = 1;
                while (true) {
                    i64 powpa = 1;
                    for (i64 _ = 0; _ < a; _++)
                        powpa *= p;  // p^a
                    if (powpa > n) break;

                    q += qpow(n, k, MOD2) - qpow(n - n / powpa, k, MOD2);
                    q %= MOD2;
                    a++;
                }
                ans = (ans * qpow(p, (q + MOD2) % MOD2, MOD)) % MOD;
            } else {
                if (!buffer.empty() && n / buffer[0] != n / p) {
                    i64 tempP = 1;
                    for (auto i : buffer) {
                        tempP = (tempP * i) % MOD;
                    }
                    i64 e = (qpow(n, k, MOD2) - qpow(n - n / buffer[0], k, MOD2) + MOD2) % MOD2;
                    ans   = (ans * qpow(tempP, e, MOD)) % MOD;
                    buffer.clear();
                }
                buffer.push_back(p);
            }

            // 这一部分是 Python
            // 代码最后多余的块逻辑，Python里清空buffer。C++不需要这么写，缓冲区逻辑只在for循环后处理即可
        }

        if (!buffer.empty()) {
            i64 tempP = 1;
            for (auto i : buffer) {
                tempP = (tempP * i) % MOD;
            }
            i64 e = (qpow(n, k, MOD2) - qpow(n - n / buffer.back(), k, MOD2) + MOD2) % MOD2;
            ans   = (ans * qpow(tempP, e, MOD)) % MOD;
            buffer.clear();
        }

        cout << ans << "\n";
    }
    return 0;
}
